### PR TITLE
31695 theme switching composer

### PR DESCRIPTION
**Pull request by lukemorawski**
### Details

This PR refactors the code to use dynamic theme/styling object, by using useThemeStyles and useTheme hooks in favor of static imports, to enable app wide theme switching.

### Fixed Issues

$ https://github.com/Expensify/App/issues/31695
PROPOSAL: no proposal

### Tests

apply this diff  to test light theme
```
diff --git a/src/styles/themes/ThemeProvider.tsx b/src/styles/themes/ThemeProvider.tsx
index 50bfb3b045..32a40cbdbf 100644
--- a/src/styles/themes/ThemeProvider.tsx
+++ b/src/styles/themes/ThemeProvider.tsx
@@ -15,7 +15,7 @@ const propTypes = {
 function ThemeProvider(props: React.PropsWithChildren) {
     const themePreference = useThemePreference();

-    const theme = useMemo(() => (themePreference === CONST.THEME.LIGHT ? lightTheme : darkTheme), [themePreference]);
+    const theme = useMemo(() => (themePreference === CONST.THEME.LIGHT ? lightTheme : lightTheme), [themePreference]);

     return <ThemeContext.Provider value={theme}>{props.children}</ThemeContext.Provider>;
 }
```
- [ ] Verify that no errors appear in the JS console

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>
no screenshots

</details>

<details>
<summary>Android: mWeb Chrome</summary>

no screenshots

</details>

<details>
<summary>iOS: Native</summary>

no screenshots

</details>

<details>
<summary>iOS: mWeb Safari</summary>

no screenshots

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

no screenshots

</details>

<details>
<summary>MacOS: Desktop</summary>

no screenshots

</details>

### CODE REVIEW

File: src/libs/ComposerUtils/updateNumberOfLines/index.native.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
 import getNumberOfLines from '@libs/ComposerUtils/getNumberOfLines';
 import updateIsFullComposerAvailable from '@libs/ComposerUtils/updateIsFullComposerAvailable';
-import styles from '@styles/styles';
+// import styles from '@styles/styles';
```
Comment (koko57) [inline src/libs/ComposerUtils/updateNumberOfLines/index.native.ts:3]: remove this line

File: src/components/EmojiPicker/EmojiPickerMenu/index.js [hunk 2]
```diff
@@ -462,7 +464,7 @@ function EmojiPickerMenu(props) {
                 />
             );
         },
-        [isUsingKeyboardMovement, highlightedIndex, onEmojiSelected, preferredSkinTone, translate, highlightFirstEmoji],
+        [preferredSkinTone, highlightedIndex, isUsingKeyboardMovement, highlightFirstEmoji, styles.emojiHeaderContainer, styles.textLabelSupporting, translate, onEmojiSelected],
```
Comment (koko57) [inline src/components/EmojiPicker/EmojiPickerMenu/index.js:467]: NAB/queestion: why did you change the order? 😃

File: src/libs/ComposerUtils/updateNumberOfLines/types.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import {NativeSyntheticEvent, TextInputContentSizeChangeEventData} from 'react-native';
 import ComposerProps from '@libs/ComposerUtils/types';
+import Styles from '@styles/styles';
 
-type UpdateNumberOfLines = (props: ComposerProps, event: NativeSyntheticEvent<TextInputContentSizeChangeEventData>) => void;
+type UpdateNumberOfLines = (props: ComposerProps, event: NativeSyntheticEvent<TextInputContentSizeChangeEventData>, styles: typeof Styles) => void;
```
Comment (koko57) [inline src/libs/ComposerUtils/updateNumberOfLines/types.ts:6]: not sure if this type is ok, @fabioh8010 wdyt?

File: src/styles/containerComposeStyles/index.native.ts [hunk 4]
```diff
@@ -1,6 +1,5 @@
-import styles from '@styles/styles';
-import ContainerComposeStyles from './types';
+import Styles from '@styles/styles';
 
-const containerComposeStyles: ContainerComposeStyles = [styles.textInputComposeSpacing];
+const containerComposeStyles = (styles: typeof Styles) => [styles.textInputComposeSpacing];
```
Comment (koko57) [inline src/styles/containerComposeStyles/index.native.ts:3]: same with the type

File: src/styles/containerComposeStyles/index.ts [hunk 5]
```diff
@@ -1,7 +1,6 @@
-import styles from '@styles/styles';
-import ContainerComposeStyles from './types';
+import Styles from '@styles/styles';
 
 // We need to set paddingVertical = 0 on web to avoid displaying a normal pointer on some parts of compose box when not in focus
-const containerComposeStyles: ContainerComposeStyles = [styles.textInputComposeSpacing, {paddingVertical: 0}];
+const containerComposeStyles = (styles: typeof Styles) => [styles.textInputComposeSpacing, {paddingVertical: 0}];
```
Comment (koko57) [inline src/styles/containerComposeStyles/index.ts:4]: same

File: src/libs/ComposerUtils/updateNumberOfLines/types.ts [hunk 6]
```diff
@@ -1,6 +1,7 @@
...
```
Comment (koko57) [inline src/libs/ComposerUtils/updateNumberOfLines/types.ts:5]: not sure if this type is ok, @fabioh8010 wdyt?

File: src/components/EmojiPicker/EmojiPickerMenu/index.js (hunk 2 above)
Comment (lukemorawski) [reply src/components/EmojiPicker/EmojiPickerMenu/index.js:467]: wasn't me, Mr. Prettier did

File: src/libs/ComposerUtils/updateNumberOfLines/index.native.ts (hunk 1 above)
Comment (lukemorawski) [reply src/libs/ComposerUtils/updateNumberOfLines/index.native.ts:3]: 🤦‍♂️

File: src/components/EmojiPicker/EmojiPickerMenu/index.js (hunk 2 above)
Comment (koko57) [reply src/components/EmojiPicker/EmojiPickerMenu/index.js:467]: 😃 ok then

File: src/styles/containerComposeStyles/index.native.ts (hunk 4 above)
Comment (lukemorawski) [reply src/styles/containerComposeStyles/index.native.ts:3]: removed it as TS infers it in a much better fashion. Original was just this, pretty blunt IMO:
```typescript
type ContainerComposeStyles = ViewStyle[];
```

File: src/styles/containerComposeStyles/index.ts (hunk 5 above)
Comment (lukemorawski) [reply src/styles/containerComposeStyles/index.ts:4]: check previous response

File: src/libs/ComposerUtils/updateNumberOfLines/types.ts (hunk 6 above)
Comment (lukemorawski) [reply src/libs/ComposerUtils/updateNumberOfLines/types.ts:5]: what's wrong here?

File: src/components/Composer/index.android.js [hunk 7]
```diff
@@ -1,10 +1,11 @@
+import {useTheme} from '@storybook/theming/dist/ts3.9/_modules/@emotion-react-types-theming';
```
Comment (rezkiy37) [inline src/components/Composer/index.android.js:1]: Wrong import.

File: src/styles/containerComposeStyles/index.ts [hunk 8]
```diff
@@ -1,7 +1,6 @@
-import styles from '@styles/styles';
-import ContainerComposeStyles from './types';
+import Styles from '@styles/styles';
```
Comment (rezkiy37) [inline src/styles/containerComposeStyles/index.ts:1]: it is not a type.
```suggestion
import styles from '@styles/styles';
```

File: src/components/Composer/index.android.js (hunk 7 above)
Comment (lukemorawski) [reply src/components/Composer/index.android.js:1]: 😱

File: src/styles/containerComposeStyles/index.native.ts [hunk 9]
```diff
@@ -1,6 +1,5 @@
-import styles from '@styles/styles';
-import ContainerComposeStyles from './types';
+import Styles from '@styles/styles';
```
Comment (rezkiy37) [inline src/styles/containerComposeStyles/index.native.ts:1]:
```suggestion
import styles from '@styles/styles';
```

File: src/styles/containerComposeStyles/index.native.ts (hunk 4 above)
Comment (rezkiy37) [reply src/styles/containerComposeStyles/index.native.ts:3]: If you decided to stop using `ContainerComposeStyles`, you should remove `types.ts` file as well.

File: src/styles/containerComposeStyles/index.ts (hunk 8 above)
Comment (lukemorawski) [reply src/styles/containerComposeStyles/index.ts:1]: but it's kind a of a module now, no?

**Review by rezkiy37 — COMMENTED** [review]
Left comments.

File: src/styles/containerComposeStyles/index.ts (hunk 8 above)
Comment (rezkiy37) [reply src/styles/containerComposeStyles/index.ts:1]: As I see it is not a module.

https://github.com/Expensify/App/blob/f5194e1d349b0f7bb80d658b44b6814f8fe065aa/src/styles/styles.ts#L3985-L3987

File: src/libs/ComposerUtils/updateNumberOfLines/types.ts (hunk 6 above)
Comment (koko57) [reply src/libs/ComposerUtils/updateNumberOfLines/types.ts:5]: shouldn't be `typeof styles` if so?

File: src/styles/containerComposeStyles/index.ts (hunk 8 above)
Comment (lukemorawski) [reply src/styles/containerComposeStyles/index.ts:1]: well, since it's philosophical question, I'll say you win.

File: src/libs/ComposerUtils/updateNumberOfLines/types.ts (hunk 6 above)
Comment (lukemorawski) [reply src/libs/ComposerUtils/updateNumberOfLines/types.ts:5]: ah, it's just a naming convention thing. Changing it now, per Micheal's request

**Comment by lukemorawski** [comment]
added some fixes

File: src/components/EmojiPicker/EmojiPickerMenu/index.js (hunk 2 above)
Comment (adhorodyski) [inline src/components/EmojiPicker/EmojiPickerMenu/index.js:467]: you can save some equality checks by adding just `styles` there - this whole object gets recreated each time and stays stable referentially for the whole time until we change the 'theme' again.

File: src/styles/containerComposeStyles/types.ts
[the remark this answers is not in the document]
Comment (VickyStash) [reply src/styles/containerComposeStyles/types.ts:1]: This way we will follow this rule: https://github.com/Expensify/App/blob/main/contributingGuides/TS_STYLE.md#file-organization

[the remark this answers is not in the document]
Comment (lukemorawski) [reply src/styles/containerComposeStyles/types.ts:1]: good thinking! Thanks

**Review by grgia — DISMISSED** [review]

File: src/styles/containerComposeStyles/index.native.ts [hunk 10]
```diff
@@ -1,6 +1,5 @@
-import styles from '@styles/styles';
 import ContainerComposeStyles from './types';
 
-const containerComposeStyles: ContainerComposeStyles = [styles.textInputComposeSpacing];
+const containerComposeStyles: ContainerComposeStyles = (stylesObject) => [stylesObject.textInputComposeSpacing];
```
Comment (chrispader) [inline src/styles/containerComposeStyles/index.native.ts:3]: can we rename this to just `styles`? :)

File: src/styles/containerComposeStyles/index.ts [hunk 11]
```diff
@@ -1,7 +1,6 @@
-import styles from '@styles/styles';
 import ContainerComposeStyles from './types';
 
 // We need to set paddingVertical = 0 on web to avoid displaying a normal pointer on some parts of compose box when not in focus
-const containerComposeStyles: ContainerComposeStyles = [styles.textInputComposeSpacing, {paddingVertical: 0}];
+const containerComposeStyles: ContainerComposeStyles = (stylesObject) => [stylesObject.textInputComposeSpacing, {paddingVertical: 0}];
```
Comment (chrispader) [inline src/styles/containerComposeStyles/index.ts:4]: same here

File: src/styles/containerComposeStyles/types.ts [hunk 12]
```diff
@@ -1,5 +1,6 @@
 import {ViewStyle} from 'react-native';
+import styles from '@styles/styles';
 
-type ContainerComposeStyles = ViewStyle[];
+type ContainerComposeStyles = (stylesObject: typeof styles) => ViewStyle[];
```
Comment (chrispader) [inline src/styles/containerComposeStyles/types.ts:4]: same here

**Review by chrispader — APPROVED** [review]
LGTM, just a few minor naming comments :)

File: src/styles/containerComposeStyles/index.native.ts (hunk 10 above)
Comment (grgia) [reply src/styles/containerComposeStyles/index.native.ts:3]: @lukemorawski would you be able to update with this change today?

**Comment by grgia** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by chrispader — APPROVED** [review]
Thanks for the changes :) LGTM

**Review by grgia — APPROVED** [review]

File: src/components/Composer/index.android.js [hunk 13]
```diff
@@ -4,7 +4,8 @@ import {StyleSheet} from 'react-native';
 import _ from 'underscore';
 import RNTextInput from '@components/RNTextInput';
 import * as ComposerUtils from '@libs/ComposerUtils';
-import themeColors from '@styles/themes/default';
+import {useTheme} from '@styles/themes/useTheme';
```
Comment (kidroca) [inline src/components/Composer/index.android.js:7]: I'm seeing the following error after I synced with main

image


Shouldn't this be 

```js
import useTheme from '@styles/themes/useTheme';
```

There doesn't seem to be a named export called `useTheme` here: [src/styles/themes/useTheme.ts](https://github.com/Expensify/App/blob/ac58fdad71b6ba542654bd0f0202995edd5850be/src/styles/themes/useTheme.ts)

Comment (arosiclair) [reply src/components/Composer/index.android.js:7]: I'm seeing the same thing. @lukemorawski why are there no testing steps and no screenshots on this PR?

Comment (situchan) [reply src/components/Composer/index.android.js:7]: > I'm seeing the same thing. @lukemorawski why are there no testing steps and no screenshots on this PR?

Maybe because QA team cannot test this PR without switching to light theme which is not enabled yet.

But it would have been better to test in all platforms though this is very simple PR. Or have C+ review and test - should the bug have been caught earlier before merge.

Comment (lukemorawski) [reply src/components/Composer/index.android.js:7]: sorry, wrong auto import! fixed now

Comment (chrispader) [reply src/components/Composer/index.android.js:7]: Is there a PR that fixes this already?

Comment (chrispader) [reply src/components/Composer/index.android.js:7]: ah perfect, thanks!

Comment (arosiclair) [reply src/components/Composer/index.android.js:7]: I understand this PR doesn't enable or change any functionality, but it does refactor theming in several components. We should always test these changes even if the test is a super simple "view each component and verify styling is still working". That would have been enough to catch small errors. QA regression testing should catch anything more complex.

@lukemorawski please don't cut corners and test on all platforms even for the simple stuff. I know it's annoying, but it's worth the time 🙇.

Comment (chrispader) [reply src/components/Composer/index.android.js:7]: should have catched this in the review as well, sorry for that! 🫠

Comment (lukemorawski) [reply src/components/Composer/index.android.js:7]: totally agree, sorry again for that
